Assets/Scripts/Player/PlayerController.cs

```diff
@@ -444,8 +445,8 @@ private void CheckCollisions()
             RaycastHit2D groundCast = CapsuleCastCollision(Vector2.down, collisionDistance);
             bool groundHit = groundCast;
             bool ceilingHit = CapsuleCastCollision(Vector2.up, collisionDistance);
-            bool leftWallHit = CapsuleCastCollision(Vector2.left, collisionDistance);
-            bool rightWallHit = CapsuleCastCollision(Vector2.right, collisionDistance);
+            bool leftWallHit = CapsuleCastCollision(Vector2.left, collisionDistance) && _velocity.x < wallVelocityThreshold;
+            bool rightWallHit = CapsuleCastCollision(Vector2.right, collisionDistance) && _velocity.x > -wallVelocityThreshold;
```

Comment:
soooo we're fine with left wall slide proccing if we have velocity 0.5 (going away)? whatever

nit: probably do the velocity check before the capsule cast check since that's more performant

but who cares it's like 0.0000001ms and it's gpu bound